### DISCUSSION TITLE
Performance vue-leaflet vs pure leaflet

### DISCUSSION THREAD

**Discussion by fractalf** [vue-leaflet/vue-leaflet#270] [Q&A] [answered]
Hello!
I'm working on an existing project that use vue-leaflet,

This is starting to be really really slow, which leaves me two choices:

1. There is just too much data for vue-leaflet to handle, so better rewrite to use native leaflet
2. The previous developer just structured everything bad, so need to refactor, still using vue-leaflet

Data:
* about 500 polygons
* about 5000 dots
* each polygon/dot is clickable and opens a modal

I guess my question is this:

**Should vue-leaflet be able to handle this amount of data smoothly?**

I believe leaflet (native) should be able to handle this flawlessly, so how much performance overhead does vue-leaflet add?

Thank you

**Comment by mikeu** [comment]
Hi @fractalf , may I ask which version of vue-leaflet you are using, and more importantly whether you see any improvement if you update to [v0.8.3](https://github.com/vue-leaflet/vue-leaflet/releases/tag/v0.8.3)? In that newly released version, the native Leaflet components are now [marked as raw](https://vuejs.org/api/reactivity-advanced.html#markraw) so that Vue no longer wraps each Leaflet object in a Proxy and makes every property on it reactive. I saw a significant speedup in the render time of a GeoJSON layer, and am hopeful that it will provide some boosts on larger datasets in general.

**Comment by fractalf** [comment]
Hm... `v0.8.3` you say?
What the hello am I using :/ 
```
"vue": "^2.6.12",
"vue2-leaflet": "^2.5.2",
```

**Comment by mikeu** [comment]
Ah, that would be the previous project, [Vue2Leaflet](https://github.com/vue-leaflet/Vue2Leaflet), and the corresponding older Vue 2. Vue2Leaflet is unlikely to be getting any major performance overhauls any time soon (unless you want to contribute one!), but this project, [vue-leaflet](https://github.com/vue-leaflet/vue-leaflet) for the current Vue 3, is still undergoing active development and improvement.

**Comment by fractalf** [comment]
@mikeu 
Ah right, ok.  
Thank you very much for clearing this up. 
I didn't set up this project myself, just got the job of "fixing" it ;)

If you would estimate very roughly the overhead of using `vue2-leaflet` vs "raw" leaflet, what would you say?

My choises are

1. Refactor the current solution to be more optimized (if thats even possible)
2. Rewrite to use raw leaflet
3. Rewrite to use vue3 + the new vue-leaflet

If I were to spend time on choise 1 it would be good to know roughly how much I could expect to gain. The current solution really is a bit of spagetti with many layers of components, so I'm pretty sure there is at least something to gain from optimizing.

I would probably spend my resources on option 2 if I knew in advance that there is not much to gain from option 1.

Unfortunatly option 3 would probably take too much of resources not available right now

**Comment by mikeu** [comment, marked as answer]
That's a pretty difficult question to answer in general! It would depend on how big the project is, how well/poorly it's been written, whether you're more familiar with or without Vue in the mix, if you're an old hand at Leaflet or learning it for the first time here, etc., etc...

That said, it might be worth spending a bit of time looking through the [Vue2Leaflet issues](https://github.com/vue-leaflet/Vue2Leaflet/issues?q=is%3Aissue) to see if you can find something similar to what you're experiencing. For example, maybe [switching to the canvas renderer](https://github.com/vue-leaflet/Vue2Leaflet/issues/473) and possibly adding the [Leaflet.Canvas-Markers](https://github.com/eJuke/Leaflet.Canvas-Markers) would give you the performance you need with almost no effort. Or if the design allows it, adding the [marker cluster plugin](https://github.com/jperelli/vue2-leaflet-markercluster) could help too, although I have no idea if those two different marker plugins play well together or interfere with each other.

From my experience and what I've seen others report, the data volume you're talking about is maybe bigger than average, but not excessive or outside the realm of what should be reasonable to handle (I guess unless those polygons each contain thousands of points maybe!).

**Comment by fractalf** [comment]
Ok, thank you very much @mikeu :)
